src/posts/en/configuring-babel-6-for-node-js.md

```diff
@@ -0,0 +1,164 @@
+
+
+### Following along the code
+Type of person who follows code instead of just reading? Code is available here in this [repo](https://github.com/abdulhannanali/babel-configuration-tutorial)
+
+### Installing and getting started with babeljs
+There's not a single way you will be able to setup Babel. Here we will be discussing enough to get up and running with it.
+
+Let's create a simple `index.js` in `code` **directory**  which will contain the following ES6 code
+```js
+function* jsRocksIsAwesome() {
+  yield "JSRocks is Awesome"
+  yield "JSRocks says JavaScript Rocks"
+  return "because JavaScript really rocks"
+}
+
+var jsRocks = jsRocksIsAwesome()
+
+console.log(jsRocks.next())
+console.log(jsRocks.next())
+console.log(jsRocks.next())
+
+```
+
+
+We'll install the **babel-cli** package by typing this command. This will install the latest stable version of **babel-cli** for the required project and list it as one of the `devDependencies` listing it in `package.json` too
+
+```bash
+npm install --save-dev babel-cli
+```
+
+Now if you run
+```bash
+babel code/index.js -d build/
+```
+
+#### Plugins and Presets
+
+You will see the same code that you wrote appear in `build/index.js`. This is where babel **plugins** and **presets** come. Babel doesn't do much on it's own, but, with **plugins** and **presets** it can do a lot. We want all the ES7 and ES6 goodness in our code.
+
+In order to do that we'll install two presets as part of our devDependencies
+- [es2015](https://babeljs.io/docs/plugins/preset-es2015/)
+- [stage-0](https://babeljs.io/docs/plugins/preset-stage-0/)
+
+Type the following command in order to install these presets.
+```bash
+npm install --save-dev babel-preset-es2015 babel-preset-stage-0
+```
+In order to access a wide range of babel plug ins click [here](https://babeljs.io/docs/plugins/)
+
+You need to include these presets in command you issue
+```bash
+babel --presets es2015,stage-0 code/index.js -o build/app.js
+```
+
+You will see normal es5 code generated in `app.js`, this is called **Transpiled code** (a term used widely in JS World). You can run this code using the command below.
+```bash
+node build/app.js
+```
+
+### Setting up a proper build environment using babel
+This is all good magic but what about doing some proper development using node.js.
+
+#### babel configuration file .babelrc
+`.babelrc` is a very neat way to separate all your babel stuff in one JSON file. It's also pretty easy to get started. This is our .babelrc file for this tutorial. You can access other .babelrc options [here](http://babeljs.io/docs/usage/options/) and make it as complex as you want it to be
+```json
+{
+  "plugins": ["es2015", "stage-0"]
+}
+```
+This is pretty much it for this tutorial, this will help us with our rest of development. Now whenever we want to add or subtract plugins. Instead of changing the command we will change the plugins array in this file. Easy! Isn't it?
```

Comment:
"this will help us with our rest of development" sounds a bit odd to me. I'm not really sure what you meant—did you mean that the Babel setup improves developer experience? I'm not sure what "this" and "rest of development" refers to in that statement.